tools/nmr_bucketing/NmrBucketing_wrapper.R

```diff
@@ -0,0 +1,307 @@
+#!/usr/local/public/bin/Rscript --vanilla --slave --no-site-file
+
+## 070115_NmrBucketing2galaxy_v1.R
+## Marie Tremblay-Franco
+## MetaboHUB: The French Infrastructure for Metabolomics and Fluxomics
+## www.metabohub.fr/en
+## marie.tremblay-franco@toulouse.inra.fr
+
+runExampleL <- FALSE
+
+if(runExampleL) {
+##------------------------------
+## Example of arguments
+##------------------------------
+argLs <- list(StudyDir = "Tlse_BPASourisCerveau",
+              upper = "10.0",
+              lower = "0.50",
+              bucket.width = "0.01",
+              exclusion = "TRUE",
+              exclusion.zone = list(c(6.5,4.5)),
+              graph="Overlay")
+
+argLs <- c(argLs,
+           list(dataMatrixOut = paste(directory,"_NmrBucketing_dataMatrix.tsv",sep=""),
+                sampleMetadataOut = paste(directory,"_NmrBucketing_sampleMetadata.tsv",sep=""),
+                variableMetadataOut = paste(directory,"_NmrBucketing_variableMetadata.tsv",sep=""),
+                graphOut = paste(directory,"_NmrBucketing_graph.pdf",sep=""),
+                logOut = paste(directory,"_NmrBucketing_log.txt",sep="")))
+}
+
+##------------------------------
+## Options
+##------------------------------
+strAsFacL <- options()$stringsAsFactors
+options(stringsAsFactors = FALSE)
+
+
+##------------------------------
+## Libraries laoding
+##------------------------------
+# For parseCommandArgs function
+library(batch)
+# For cumtrapz function
+library(pracma)
+
+# R script call
+source_local <- function(fname)
+{
+	argv <- commandArgs(trailingOnly = FALSE)
+	base_dir <- dirname(substring(argv[grep("--file=", argv)], 8))
+	source(paste(base_dir, fname, sep="/"))
+}
+#Import the different functions
+source_local("NmrBucketing_script.R")
+source_local("DrawSpec.R")

```

Comment:
I think it would be easier to define one function in a file which takes all arguments, then call source on that file in the galaxy wrapper and then call that function. This would remove the entire dependency and command line passing logic. Another options would be to use configfiles.